introspection.go

```diff
@@ -128,11 +140,41 @@ func (ch *Channel) IntrospectState(opts *IntrospectionOptions) *RuntimeState {
 	ch.mutable.RUnlock()
 
 	return &RuntimeState{
+		CreatedStack:   ch.createdStack,
 		LocalPeer:      ch.PeerInfo(),
 		SubChannels:    ch.subChannels.IntrospectState(opts),
 		RootPeers:      ch.rootPeers().IntrospectState(opts),
 		Peers:          ch.Peers().IntrospectList(opts),
 		NumConnections: conns,
+		OtherChannels:  ch.IntrospectOthers(opts),
+	}
+}
+
+// IntrospectOthers returns the OtherChannelState for all other channels in this process.
+func (ch *Channel) IntrospectOthers(opts *IntrospectionOptions) map[string][]OtherChannelState {
+	channelMap.Lock()
+	defer channelMap.Unlock()
+
+	states := make(map[string][]OtherChannelState)
+	for svc, channels := range channelMap.existing {
+		otherChannelStates := make([]OtherChannelState, 0, len(channels))
+		for _, otherChan := range channels {
+			if ch == otherChan {
+				continue
+			}
+			otherChannelStates = append(otherChannelStates, otherChan.IntrospectOtherState(opts))
+		}
+		states[svc] = otherChannelStates
+	}
+
+	return states
+}
+
+// IntrospectOtherState returns OtherChannelState for a channel.
+func (ch *Channel) IntrospectOtherState(opts *IntrospectionOptions) OtherChannelState {
```

Comment:
Unfortunately `ChannelState` is already used for the current channel's state enum (closed, active etc). I've renamed it to `ChannelInfo` and `ReportInfo`.